infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironmentFactory.java

```diff
@@ -124,6 +124,8 @@ protected KubernetesEnvironment doCreate(
       checkNotNull(object.getMetadata(), "%s metadata must not be null", object.getKind());
       checkNotNull(object.getMetadata().getName(), "%s name must not be null", object.getKind());
 
+      object.getMetadata().setNamespace(null);
```

Comment:
Please add a few words that it is needed because it is set to Che Server one by K8s API during list loading

Reply:
ok